### PR TITLE
Http handle invalid byte sequence

### PR DESCRIPTION
**Pull request by picatz**
HTTP requests and responses are quite a nuisance to deal with. To be quite honest, I'm a little worried about what I might have missed implementing this.

I'm sure I'll stumble upon those problems ( so I can fix them ) later. 😂

Aims to fix https://github.com/sdaubert/packetgen/issues/78

### CODE REVIEW

File: lib/packetgen/header/http/request.rb [hunk 1]
```diff
@@ -85,6 +87,6 @@ def to_s
     end
 
     self.add_class HTTP::Request 
-    TCP.bind_header HTTP::Request, body: ->(b) { /^(CONNECT|DELETE|GET|HEAD|OPTIONS|PATCH|POST|PUT)/ =~ b }
+    TCP.bind_header HTTP::Request, body: ->(b) { /(CONNECT|DELETE|GET|HEAD|OPTIONS|PATCH|POST|PUT)/ =~ b.chars.select(&:valid_encoding?).join }
```
Comment (sdaubert) [inline lib/packetgen/header/http/request.rb]: Why not using HTTP::VERBS to construct the regexp? This should avoid repeating yourself.
Furthermore, the regexp may be built in same module than VERBS and may be a constant.

Comment (sdaubert) [inline lib/packetgen/header/http/request.rb]: Why did you remove the `^` in front of regexp?

**Comment by sdaubert** [comment]
OK, nice PR.

Please add a spec to check #81 is fixed. I will merge it next week.

File: lib/packetgen/header/http/request.rb (hunk 1 above)
Comment (picatz) [reply lib/packetgen/header/http/request.rb]: This is a good point 👍  

This was left over from the initial implementation. I will try to fix this.

Comment (picatz) [reply lib/packetgen/header/http/request.rb]: From what I've seen in some packets: the actual TCP body may not actually start that way, but still be a valid HTTP body to parse.

Now that you mention it: I will go over this again to try and provide a better solution.

**Comment by sdaubert** [comment]
@picatz please add a spec to check #81 is fixed, and i will merge this.

**Comment by picatz** [comment]
@sdaubert Will happen this week 👍

**Comment by picatz** [comment]
@sdaubert I got the spec implemented, and while doing so, fixed some issues I found. 😂
